plotting/plot_factory/trace.py

```diff
@@ -0,0 +1,81 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Constructs a plot and DashApp object for insertion into directly into a web page
+"""
+
+# Core Dependencies
+import sys
+import plotly.graph_objects as go  # pylint: disable=unused-import
+
+
+class Trace:  # pylint: disable=too-many-arguments, line-too-long
+    """Creates a trace object """
+    def __init__(self, data, plot_style, plot_name, mode=None, error_bars=None):  # pylint: disable=too-many-arguments, line-too-long
+        """
+        Trace Object
+        :param data (pandas dataframe)
+        :param plot_style (dictionary)
+        :param plot_name (string)
+        :param mode (string)
+        :param error_bars (bool)
+        """
+
+        self.trace = self.create_trace(mode=mode,
+                                       data=data,
+                                       plot_style=plot_style,
+                                       name=plot_name,
+                                       error_bars=error_bars)
+
+    @staticmethod
+    def trace_dict(data, error_bars):
+        """
+        Creates trace dictionary
+
+        :param data (dataframe)
+        :param error_bars (bool)
+
+        :return: trace (dictionary)
+        """
+        trace = {}
+        for axis in list(data.columns):
+            if axis == 'E':
+                trace['error_y'] = dict(type='data',
+                                        array=data[axis].to_list(),
+                                        visible=error_bars)
+            else:
+                trace[axis.lower()] = data[axis]
+        return trace
+    # pylint disable=trailing-whitespace
```

Comment:
Where is the trailing whitespace here?

Reply:
Not sure, Travis kept picking it up and failing builds despite no trailing spaces being present. Adding the disable seemed to fix the problem. 😕 
I have since removed the pylint disable after amending other areas within trace.py and Travis seems to be fine now